packages/auth-construct/src/construct.ts

```diff
@@ -900,132 +904,194 @@ export class AmplifyAuth
-        requirements.push('REQUIRES_SYMBOLS');
-      }
+    // extract the passwordPolicy from the UserPool policies
+    output.passwordPolicyMinLength = Lazy.string({
+      produce: () => {
+        if (!cfnUserPool.policies) {
+          return;
+        }
+        const policy = (cfnUserPool.policies as CfnUserPool.PoliciesProperty)
+          .passwordPolicy as CfnUserPool.PasswordPolicyProperty;
+        return policy.minimumLength?.toString();
+      },
+    });
 
-      if (requirements.length > 0) {
-        output.passwordPolicyRequirements = JSON.stringify(requirements);
-      }
-    }
+    output.passwordPolicyRequirements = Lazy.string({
+      produce: () => {
+        if (!cfnUserPool.policies) {
+          return;
+        }
+        const policy = (cfnUserPool.policies as CfnUserPool.PoliciesProperty)
+          .passwordPolicy as CfnUserPool.PasswordPolicyProperty;
+        const requirements = [];
+        policy.requireNumbers && requirements.push('REQUIRES_NUMBERS');
+        policy.requireLowercase && requirements.push('REQUIRES_LOWERCASE');
+        policy.requireUppercase && requirements.push('REQUIRES_UPPERCASE');
+        policy.requireSymbols && requirements.push('REQUIRES_SYMBOLS');
+        return JSON.stringify(requirements);
+      },
+    });
 
-    if (this.computedUserPoolProps.mfa) {
-      output.mfaConfiguration = this.computedUserPoolProps.mfa;
+    // extract the MFA configuration setting from the UserPool resource
+    output.mfaConfiguration = Lazy.string({
+      produce: () => {
+        return cfnUserPool.mfaConfiguration ?? 'OFF';
+      },
+    });
+    // extract the MFA types from the UserPool resource
+    output.mfaTypes = Lazy.string({
+      produce: () => {
+        const mfaTypes: string[] = [];
+        (cfnUserPool.enabledMfas ?? []).forEach((type) => {
+          if (type === 'SMS_MFA') {
+            mfaTypes.push('SMS');
+          }
+          if (type === 'SOFTWARE_TOKEN_MFA') {
+            mfaTypes.push('TOTP');
+          }
+        });
+        return JSON.stringify(mfaTypes);
+      },
+    });
 
-      const mfaTypes = [];
-      if (this.computedUserPoolProps.mfaSecondFactor?.otp) {
-        mfaTypes.push('TOTP');
-      }
-      if (this.computedUserPoolProps.mfaSecondFactor?.sms) {
-        mfaTypes.push('SMS');
-      }
-      if (mfaTypes.length > 0) {
-        output.mfaTypes = JSON.stringify(mfaTypes);
-      }
-    }
+    // extract social providers from UserPool resource
+    output.socialProviders = Lazy.string({
+      produce: () => {
+        const outputProviders = [];
+        const userPoolProviders = this.resources.userPool.identityProviders;
+        if (!userPoolProviders || userPoolProviders.length === 0) {
+          return;
+        }
+        for (const provider of userPoolProviders) {
+          const providerResource = provider.node.findChild(
+            'Resource'
+          ) as CfnUserPoolIdentityProvider;
```

Comment:
We should add type checks for all places where we do `.node.findChild`

Reply:
updated